metaworld/policies/mujoco/sawyer_xyz/base.py

```diff
@@ -0,0 +1,67 @@
+import abc
+import warnings
+
+import numpy as np
+
+from metaworld.policies.base import Policy
+
+
+class SawyerXYZPolicy(Policy, metaclass=abc.ABCMeta):
+    def __init__(self, pid_xyz):
+        self._pid_xyz = pid_xyz
+
+    def get_action(self, obs):
+        """Gets an action in response to an observation.
+
+        Args:
+            obs (np.ndarray): Observation which conforms to env.observation_space
+
+        Returns:
+            np.ndarray: 4 element array representing the action to take
+        """
+        xyz_curr = obs[self.obs_idx_of_effector_xyz]
+        xyz_error = self.xyz_goal(obs) - xyz_curr
+        xyz_response = self._pid_xyz.response_to(xyz_error)
+
+        if np.any(np.absolute(xyz_response) > 1.):
+            warnings.warn('PID constants may be too high. Response gets clipped at [-1, 1]')
+
+        action = np.zeros(4)
+        action[self.act_idx_of_effector_xyz] = xyz_response
+        action[self.act_idx_of_effector_grab] = self.grab_amount(obs)
+        return action
+
+    @abc.abstractmethod
+    def xyz_goal(self, obs):
+        raise NotImplementedError
```

Comment:
It's not possible to instantiate this class without implementing this method. You can just leave it blank.